Jenkinsfile

```diff
@@ -0,0 +1,85 @@
+pipeline {
+    options {
+        disableConcurrentBuilds()
+        timestamps()
+    }
+    agent {
+        node { label 'master' }
+    }
+    environment {
+        GITHUB_CREDENTIALS = credentials('brave-builds-github-token-for-pr-builder')
+        BC_REPO = "https://${GITHUB_CREDENTIALS}@github.com/brave/brave-browser"
+        BC_BRANCH = "brave-core-${GIT_BRANCH}"
+    }
+    stages {
+        stage('checkout') {
+            steps {
+                echo "BRANCH_NAME: ${BRANCH_NAME}"
+                echo "CHANGE_TARGET: ${CHANGE_TARGET}"
+                echo "CHANGE_ID: ${CHANGE_ID}"
+                echo "CHANGE_TITLE: ${CHANGE_TITLE}"
+                echo "CHANGE_AUTHOR: ${CHANGE_AUTHOR}"
+                echo "CHANGE_AUTHOR_DISPLAY_NAME: ${CHANGE_AUTHOR_DISPLAY_NAME}"
+                sh """
+                    if [ -d brave-browser ]; then
+                        rm -rf brave-browser
+                    fi
+
+                    git clone -b ${CHANGE_TARGET} ${BC_REPO}
+
+                    cd brave-browser
+
+                    if [ -z "`git ls-remote --heads ${BC_REPO} ${BC_BRANCH}`" ]; then
+                        git checkout -f -b ${BC_BRANCH}
+                    else
+                        git checkout -f ${BC_BRANCH}
+                    fi
+                """
+            }
+        }
+        stage('config') {
+            steps {
+                sh """
+                    git config -f brave-browser/.git/config user.name brave-builds
+                    git config -f brave-browser/.git/config user.email devops@brave.com
+                    git config -f brave-browser/.git/config push.default simple
+                """
+            }
+        }
+        stage('pin') {
+            steps {
+                sh """
+                    jq "del(.config.projects[\\"brave-core\\"].branch) | .config.projects[\\"brave-core\\"].commit=\\"${GIT_COMMIT}\\"" brave-browser/package.json > brave-browser/package.json.new
+                    mv brave-browser/package.json.new brave-browser/package.json
+                """
+            }
+        }
+        stage('push') {
+            steps {
+                sh """
+                    git -C brave-browser commit -a -m "pin brave-core to ${GIT_COMMIT} from ${GIT_BRANCH}"
+                    git -C brave-browser push ${BC_REPO}
+                """
+            }
+        }
+        stage('sleep') {
+            steps {
+                sleep time: 5, unit: 'MINUTES'
```

Comment:
Any reason for the arbitrary 5 minute time amount? Should we could make this configurable (passed into job) or override-able?

Reply:
so those 5 minutes are necessary for the other pipeline in `brave-browser` can discover the new branch and create a Jenkins job for it that this `brave-core` pipeline can call

Reply:
ah - makes sense. Since it queues up the https://staging.ci.brave.com/job/brave-core-build-pr job

Maybe instead of `sleep`, we can call it `wait_for_branch_propogation` (or something like this)

Reply:
oh that's a jenkins keyword, `sleep`

Reply:
tried to keep them one word so it doesn't break the formatting of the pipeline, can probably put `wait` instead